packages/data-stores/src/plans/queries/use-plans.ts

```diff
@@ -14,14 +15,15 @@ interface PlansIndex {
  */
 function usePlans( { coupon }: { coupon: string | undefined } ): UseQueryResult< PlansIndex > {
 	const queryKeys = useQueryKeysFactory();
+	const locale = useLocale();
 	const params = new URLSearchParams();
 	coupon && params.append( 'coupon_code', coupon );
 
 	return useQuery( {
 		queryKey: queryKeys.plans( coupon ),
 		queryFn: async (): Promise< PlansIndex > => {
 			const data: PricedAPIPlan[] = await wpcomRequest( {
-				path: `/plans`,
+				path: `/plans?locale=${ locale }`,
```

Comment:
Nit: To avoid the string interpolation, it can be be appended to the query params, similarly to how the `coupon_code` is added:

```js
params.append( 'locale', locale );
```